source/language/types.rst

```diff
@@ -135,8 +124,63 @@ circuits. Physical qubits must not be declared and they are, as all the qubits,
    qubit γ;
    // Declare a qubit register with 20 qubits
    qubit[20] qubit_array;
+
+.. _physical-qubits:
+
+Physical Qubits
+~~~~~~~~~~~~~~~
+
+Physical qubits refer to particular hardware qubits. Therefore, they are only fully defined with
+respect to a target device that has a published device topology. The hardware provider determines
+the integer-labels associated with each qubit within the device's topology.
+
+While virtual qubits can be named, hardware qubits are referenced by the syntax ``$[NUM]``, where
+``NUM`` is a non-negative integer. Any integer included in the published device topology is a valid
+physical qubit identifier. Note that this implies that physical qubit identifier indices may be
+non-consecutive, depending on the device.
+
+Like virtual qubits, physical qubits are global variables, but unlike virtual qubits, they must
+not be declared.
+
+These qubit types are often used for ``defcal`` blocks because calibrations are typically valid
+only for a particular set of physical qubits (see also :ref:`pulse gates <pulse-gates>`).
+
+.. TODO: Where is a better place for physical circuit discussion?
+.. TODO: Should we introduce executable circuit as another term for physical circuit?
+
+Physical qubits are also used in lower parts of the compilation stack when emitting physical
+circuits. A physical circuit is one which only references physical qubits, and every operation
+used in the circuit has an associated ``defcal``, which we can call hardware-native gates or
+measurements. A physical circuit can be directly executed on the target hardware.
+
+.. code-block::
+
    // CNOT gate between physical qubits 0 and 1
    CX $0, $1;
+   // Define the pulse-level instruction sequence for ``h`` on physical qubit 0
+   defcal h $0 { ... }
+
+Physical qubit constraints
+..........................
+.. TODO: Can physical qubits be used in a ``defgate``? My answer would be no.
```

Comment:
Update: Erik gave a good argument for why it would be possible, so I'm inclined to revise this.